keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/DebianArtifactSupplier.kt

```diff
@@ -28,23 +28,24 @@ class DebianArtifactSupplier(
   override val eventPublisher: EventPublisher,
   private val artifactService: ArtifactService,
   override val artifactMetadataService: ArtifactMetadataService
-) : BaseArtifactSupplier<DebianArtifact, DebianVersioningStrategy>(artifactMetadataService) {
+) : BaseArtifactSupplier<DebianArtifact, DebianVersionSortingStrategy>(artifactMetadataService) {
   private val log by lazy { LoggerFactory.getLogger(javaClass) }
 
   override val supportedArtifact = SupportedArtifact("deb", DebianArtifact::class.java)
 
-  override val supportedVersioningStrategy =
-    SupportedVersioningStrategy("deb", DebianVersioningStrategy::class.java)
+  override val supportedSortingStrategy =
+    SupportedSortingStrategy("deb", DebianVersionSortingStrategy::class.java)
 
   override fun getLatestArtifact(deliveryConfig: DeliveryConfig, artifact: DeliveryArtifact): PublishedArtifact? =
     runWithIoContext {
       artifactService.getVersions(artifact.name, artifact.statusesForQuery, DEBIAN)
-        .map { version -> "${artifact.name}-$version" }
-        .sortedWith(artifact.versioningStrategy.comparator)
-        .firstOrNull() // versioning strategies return descending by default... ¯\_(ツ)_/¯
-        ?.let { version ->
-          artifactService.getArtifact(artifact.name, version.removePrefix("${artifact.name}-"), DEBIAN)
+        // FIXME: this is making N calls to fill in data for each version so we can sort.
```

Comment:
For the time being, I've worked around the lack of support in the Rocket API for returning a list of artifact versions with details by adding an additional call to Igor/Rocket for each version returned. I realize this is expensive an technically unnecessary, but as discussed in chat, my focus at the moment is to get something working, which we can improve for performance later.

Reply:
👍 I think this is fine for now until we have the correct API.

Reply:
where are the calls are being made?

Reply:
On line 48:
```kotlin
artifactService.getArtifact(artifact.name, version, DEBIAN)
```

Reply:
maybe i'm missing something - where are you calling this N times? I see only 2 calls - one to fetch all versions. and one to get the actual artifact data.

Reply:
Right, but the second call is in a loop.